jsonschema/infer.go

```diff
@@ -125,7 +126,12 @@ func forType(t reflect.Type, seen map[reflect.Type]bool, ignore bool, schemas ma
 	}
 
 	if s := schemas[t]; s != nil {
-		return s.CloneSchemas(), nil
+		cloned := s.CloneSchemas()
+		if os.Getenv("JSONSCHEMAGODEBUG") != "typeschemasnull=1" && allowNull && s.Type != "" {
```

Comment:
No need to parse the environment variable, but please define a global constant with the variable name.